extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -256,9 +269,89 @@ public void build(RecorderContext recorderContext, HibernateOrmRecorder recorder
                     .add((Class<? extends ServiceContributor>) recorderContext.classProxy(serviceContributorClassName));
         }
 
+        Set<String> entitiesToGenerateProxiesFor = new HashSet<>(domainObjects.getEntityClassNames());
+        for (ParsedPersistenceXmlDescriptor unit : allDescriptors) {
+            entitiesToGenerateProxiesFor.addAll(unit.getManagedClassNames());
+        }
+
+        PreGeneratedProxies proxyDefinitions = generatedProxies(entitiesToGenerateProxiesFor, compositeIndex,
```

Comment:
I know it's horrible, but given the context I think I'd prefer this method to be surrounded by a catch Runtime exception / log warning ?

That's to ensure that any unforeseen issue with those proxy generation will indeed get us the "falllback the current state" behaviour we discussed.

(Especially as the method intentionally throws RuntimeException on any CNFE)

Reply:
p.s. as a fallback of course register an empty `PreGeneratedProxies`

Reply:
Why does this get such special treatment? Any PR that normally included such a try/catch because the author was not sure if would work would get rejected. If there is an NPE or some other problem then IMHO this is no different to any other bug.

Reply:
@Sanne maybe we should ask for broader review from the hibernate team if you are worried there are cases I have missed?

Reply:
yes I would generally react like you say. but we decided to go very cautious here and promised that _any problem_ this would fallback to the previous, slower behaviour gracefully.

As you prefer though, I don't consider it blocking.

Reply:
> @Sanne maybe we should ask for broader review from the hibernate team if you are worried there are cases I have missed?

because we either will get a timely answer but not entirely sure, or it will take much longer.

Reply:
Against my better judgement I have added it.

Reply:
Created https://github.com/quarkusio/quarkus/issues/8447 to remove it in 1.5

Reply:
thank you both!

Reply:
> Created #8447 to remove it in 1.5

why not in 1.4 ?